DataModel/Certificate.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataModel
+{
+    public class Certificate : CompetencyType
```

Comment:
Are you sure you want to be inheriting CompetencyType here? That will put all the certificates in the same table as competencytypes. You can see the details of inheritance with EF Core here: 
Inheritance in EF Core  can be seen here: https://docs.microsoft.com/en-us/ef/core/modeling/inheritance

In generally, I find inheritance something you can avoid using.

Reply:
Certificates are defined as a type of competence according to the specifications. It has attributes common to that of a competence, with the exception of Rating. We considered heritage to be the way to implement this relationship.

Reply:
Could we use table-per-inheritance instead of the Entity type hierarchy? 
https://docs.microsoft.com/en-us/ef/core/modeling/inheritance#table-per-type-configuration

I find the Entity type hierarchy very confusing if you ever have to query the data with SQL

Reply:
That's unfortunate about the inhertiance type not being support until verison 5. Here's a page that goes over another solution that should work:
https://www.thinktecture.com/en/entity-framework-core/table-per-type-inheritance-support-part-1-code-first/